include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -482,6 +482,7 @@ typedef struct OrtTensorRTProviderOptions {
   int trt_engine_decryption_enable;             // enable engine decryption. Default 0 = false, nonzero = true
   const char* trt_engine_decryption_lib_path;   // specify engine decryption library path
   int trt_force_sequential_engine_build;        // force building TensorRT engine sequentially. Default 0 = false, nonzero = true
+  int trt_cuda_graph_enable;                    // enable cuda graph. Default 0 = false, nonzero = true
```

Comment:
Shouldn't the struct get versioned with the addition of a new option ?

Reply:
I thought we agreed we shouldn't be updating provider structs in c api anymore (for the very reason that Hari brings up about versioning)
and instead only updating the opaque struct OrtTensorRTProviderOptionsV2
+@chilo-ms FYI

Reply:
Yeah, I think it should become OrtTensorRTProviderOptionsV3 if V2 has shipped with the previous ORT release (This was my understanding)

Reply:
> Yeah, I think it should become OrtTensorRTProviderOptionsV3 if V2 has shipped with the previous ORT release (This was my understanding)

we don't need to update the version of the opaque struct when adding fields right? since it's only accessed via api and not directly. if the newly added field can't be represented as a string, then we would need to add another api to access those.

Reply:
IIRC from the time we were discussing it, we need to version it for any mutation (addition or removal of fields). If we add support for a new field without versioning it, doesn't the UpdateTensorRTProviderOptions API behave differently in ORT 1.10 (where the V2 struct won't support the new field) and in ORT 1.11 (where the V2 struct will support the new field) ?

Reply:
> IIRC from the time we were discussing it, we need to version it for any mutation (addition or removal of fields). If we add support for a new field without versioning it, doesn't the UpdateTensorRTProviderOptions API behave differently in ORT 1.10 (where the V2 struct won't support the new field) and in ORT 1.11 (where the V2 struct will support the new field) ?

versioning the struct was needed originally if the user directly manipulates the struct.
I thought having an opaque struct behind an api gives us flexibility for not having to change the struct every time.
let's revisit this discussion offline. maybe i'm misunderstanding something in this case.

Reply:
We had a consensus on using key-value strings for provider options that can be represented by string but we don't explicitly say we support versioning. But we require error reporting for undocumented config keys.

@stevenlix, here is my ["enable timing chach" PR](https://github.com/microsoft/onnxruntime/pull/10297), you can reference it to add new filed to opaque struct OrtTensorRTProviderOptionsV2

Reply:
No versioning of struct is required beyond V2. The availability or the unavailability of APIs to manipulate the V2 struct provides the versioning.

Reply:
"The availability or the unavailability of APIs to manipulate the V2 struct provides the versioning." - What APIs exists to manipulate the V2 struct will be the same in ORT 1.10 and ORT 1.11 won't they ? It is just that the `UpdateTensorRTProviderOptions()` API will additionally support one more key (enable_cuda_graph) in 1.11 (which obviously won't be supported in the released 1.10).

Reply:
yes, one uses CreateTensorRTProviderOptions(), UpdateTensorRTProviderOptions() to create and update the struct.
The second api deals with strings, and will recognize a new string key "enable_cuda_graph" in ort 1.11
api signatures don't change.

Reply:
talked to @chilo-ms offline. He is going to make a separate PR from the timing cache PR for OrtTensorRTProviderOptionsV2. After the PR merged, I will add cuda_graph option in OrtTensorRTProviderOptionsV2.